### PR TITLE
Room persistency - Recents

### PR DESCRIPTION
**Pull request by kordianbruck**
please review <3
#560 

Please merge after #614

### CODE REVIEW

File: app/src/main/java/de/tum/in/tumcampusapp/activities/PersonsSearchActivity.java [hunk 1]
```diff
@@ -74,6 +58,20 @@ public void onCreate(Bundle savedInstanceState) {
...
 
+    private ArrayList<Person> getRecents() {
+        List<Recent> recentList = recentsManager.getAllFromDb();
+        ArrayList<Person> personList = new ArrayList<>(recentList.size());
+        for (Recent r : recentList) {
+            String[] t = r.getName()
+                          .split("\\$");
+            Person p = new Person();
+            p.setId(t[0]);
+            p.setName(t[1]);
```
Comment (pfent) [inline app/src/main/java/de/tum/in/tumcampusapp/activities/PersonsSearchActivity.java]: When you're at it:
This should really be extracted to a function `Person.fromRecent()`, so we can maybe use this in a Kotlin `Iterable.map(Person.fromRecent(it))`

File: app/src/main/java/de/tum/in/tumcampusapp/activities/PersonsSearchActivity.java [hunk 2]
```diff
@@ -93,26 +91,7 @@ public void onItemClick(AdapterView<?> a, View v, int position, long id) {
...
-                    String recent = personsCursor.getString(0);
-                    String[] t = recent.split("\\$");
-                    Person p = new Person();
-                    p.setId(t[0]);
-                    p.setName(t[1]);
-                    list.add(p);
-                } while (personsCursor.moveToNext());
-            }
-        }
-        ListAdapter adapter = new ArrayAdapter<>(this, android.R.layout.simple_list_item_1, android.R.id.text1, list);
```
Comment (pfent) [inline app/src/main/java/de/tum/in/tumcampusapp/activities/PersonsSearchActivity.java]: Removing duplicated code :tada:

File: app/src/main/java/de/tum/in/tumcampusapp/activities/RoomFinderActivity.java [hunk 3]
```diff
@@ -125,21 +125,17 @@ private void openRoomDetails(Serializable room) {
...
+        List<Recent> recentList = recentsManager.getAllFromDb();
+        List<RoomFinderRoom> roomList = new ArrayList<>(recentList.size());
+        for (Recent r : recentList) {
+            String[] values = r.getName()
+                               .split(";");
+            if (values.length != 6) {
+                continue;
             }
+            RoomFinderRoom room = new RoomFinderRoom(values[0], values[1],
+                                                     values[2], values[3], values[4], values[5]);
```
Comment (pfent) [inline app/src/main/java/de/tum/in/tumcampusapp/activities/RoomFinderActivity.java]: Plz extract to function `RoomFinderRoom.fromRecent()`

File: app/src/main/java/de/tum/in/tumcampusapp/api/TUMCabeClient.java [hunk 4]
```diff
@@ -83,23 +83,6 @@ private TUMCabeClient(final Context c) {
...
-                if (t instanceof SSLPeerUnverifiedException) {
-                    //TODO show a error message
-                    //Toast.makeText(context, t.toString(), Toast.LENGTH_LONG).show();
-                }
-
-                //Return the same cause, so it can be handled by other activities
-                return cause;
-            }
-        }; */
-
```
Comment (pfent) [inline app/src/main/java/de/tum/in/tumcampusapp/api/TUMCabeClient.java]: Has the error handler been implemented?
If not, please leave the TODO

File: app/src/main/java/de/tum/in/tumcampusapp/managers/TransportManager.java [hunk 5]
```diff
@@ -372,4 +369,20 @@ public void onRequestCard(Context context) {
...
+
+
+    public static List<StationResult> getRecentStations(RecentsManager recentsManager) {
+        List<Recent> recentList = recentsManager.getAllFromDb();
+        List<StationResult> stationResults = new ArrayList<>(recentList.size());
+        for (Recent r : recentList) {
+            try {
+                String jsonStationResult = r.getName();
+                StationResult stationResult = gson.fromJson(jsonStationResult, StationResult.class);
+                stationResults.add(stationResult);
```
Comment (pfent) [inline app/src/main/java/de/tum/in/tumcampusapp/managers/TransportManager.java]: `StationResult.fromRecent()`

File: app/src/main/java/de/tum/in/tumcampusapp/models/tumcabe/WifiMeasurement.kt [hunk 6]
```diff
@@ -1,9 +1,23 @@
...
                            var bssid: String = "",
                            var dBm: Int = -1,
                            var accuracyInMeters: Float = -1f,
                            var latitude: Double = -1.0,
-                           var longitude: Double = -1.0)
+                           var longitude: Double = -1.0) {
+    companion object {
+        fun create(ssid: String, bssid: String, dBm: Int, accuracyInMeters: Float, latitude: Double, longitude: Double): WifiMeasurement {
+            val now = Utils.getDateTimeString(Date())
+            return WifiMeasurement(date = now, ssid = ssid, bssid = bssid, dBm = dBm, accuracyInMeters = accuracyInMeters, latitude = latitude, longitude = longitude)
```
Comment (pfent) [inline app/src/main/java/de/tum/in/tumcampusapp/models/tumcabe/WifiMeasurement.kt]: Why not directly do this in the declaration?
`var date: String = Utils.getDateTimeString(Date())`

File: app/src/main/java/de/tum/in/tumcampusapp/api/TUMCabeClient.java (hunk 4 above)
Comment (kordianbruck) [reply app/src/main/java/de/tum/in/tumcampusapp/api/TUMCabeClient.java]: As of retrofit 2, network errors throw a normal ioexception. There is no need to have a central error handler, as previously.

If we really do want to logg this somewhere central, you would need to add an interceptor to the OkHttpClient - we already have one there.

**Comment by kordianbruck** [comment]
@pfent done.

**Comment by pfent** [comment]
Even with proper Exception handling now. Thx

**Comment by kordianbruck** [comment]
@pfent yups. Also: please make use of squash & merge :+1:

**Comment by pfent** [comment]
@kordianbruck I did squash :thinking:

**Comment by kordianbruck** [comment]
Then github has a bad UI for that:
" pfent merged commit 97efcc4 into master  an hour ago"

**Pull request merged**